src/test/script_p2sh_tests.cpp

```diff
@@ -201,7 +201,7 @@ BOOST_AUTO_TEST_CASE(set)
     {
         SignatureData empty;
         BOOST_CHECK_MESSAGE(SignSignature(keystore, CTransaction(txFrom), txTo[i], 0, SIGHASH_ALL, empty), strprintf("SignSignature %d", i));
-        BOOST_CHECK_MESSAGE(IsStandardTx(CTransaction(txTo[i]), reason), strprintf("txTo[%d].IsStandard", i));
+        BOOST_CHECK_MESSAGE(i != IsStandardTx(CTransaction(txTo[i]), reason), strprintf("txTo[%d].IsStandard", i));
```

Comment:
??? this change makes no sense to me. `i` is an `int`, but `IsStandardTx` returns a bool...

Reply:
See b90ef78f7ab564fb5948583e7a5b5399f9798301 for how I fixed this in Knots

Reply:
`(i == 1) == IsStandardTx(..)` would probably be clearer then?

Reply:
With this current code, for `i>1` it doesn't matter what `IsStandardTx` returns, the `BOOST_CHECK` will succeed either way.

Here's a patch that just tests both cases so it doesn't depend on the value of `DEFAULT_PERMIT_BAREMULTISIG`: https://github.com/ajtowns/bitcoin/commit/1c139da6e8d697df419ca40642cb52ca17bdc526